test/common/grpc/grpc_web_filter_test.cc

```diff
@@ -0,0 +1,126 @@
+#include "common/buffer/buffer_impl.h"
+#include "common/common/base64.h"
+#include "common/grpc/grpc_web_filter.h"
+#include "common/http/header_map_impl.h"
+#include "common/http/headers.h"
+
+#include "test/mocks/http/mocks.h"
+#include "test/mocks/upstream/mocks.h"
+#include "test/test_common/printers.h"
+#include "test/test_common/utility.h"
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+using testing::_;
+using testing::NiceMock;
+using testing::Invoke;
+
+namespace Envoy {
+namespace Grpc {
+namespace {
+const char MESSAGE[] = "\x00\x00\x00\x00\x11grpc-web-bin-data";
+const size_t MESSAGE_SIZE = 22;
+const char TEXT_MESSAGE[] = "\x00\x00\x00\x00\x12grpc-web-text-data";
+const size_t TEXT_MESSAGE_SIZE = 23;
+const char B64_MESSAGE[] = "AAAAABJncnBjLXdlYi10ZXh0LWRhdGE=";
+const size_t B64_MESSAGE_SIZE = 32;
+const char TRAILERS[] = "\x80\x00\x00\x00\x0fgrpc-status:0\r\n";
+const size_t TRAILERS_SIZE = 20;
+} // namespace
+
+class GrpcWebFilterTest : public testing::Test {
+public:
+  GrpcWebFilterTest() : filter_() { filter_.setEncoderFilterCallbacks(encoder_callbacks_); }
+
+  ~GrpcWebFilterTest() override {}
+
+  GrpcWebFilter filter_;
+  NiceMock<Http::MockStreamEncoderFilterCallbacks> encoder_callbacks_;
+};
+
+TEST_F(GrpcWebFilterTest, BinaryUnary) {
+  // Tests request headers.
+  Http::TestHeaderMapImpl request_headers;
+  request_headers.addViaCopy(Http::Headers::get().ContentType,
+                             Http::Headers::get().ContentTypeValues.GrpcWeb);
+  EXPECT_EQ(Http::FilterHeadersStatus::Continue, filter_.decodeHeaders(request_headers, false));
+  EXPECT_EQ(Http::Headers::get().ContentTypeValues.Grpc,
+            request_headers.ContentType()->value().c_str());
+
+  // Tests request data.
+  Buffer::OwnedImpl request_buffer(MESSAGE, MESSAGE_SIZE);
+  EXPECT_EQ(Http::FilterDataStatus::Continue, filter_.decodeData(request_buffer, true));
+  EXPECT_EQ(0, memcmp(MESSAGE, TestUtility::bufferToString(request_buffer).c_str(), MESSAGE_SIZE));
+
```

Comment:
I think you should still verify this pass-thru behavior, as it's something we intend to have happen and don't want to see a regression for.